dashboard/views_test.py

```diff
@@ -153,3 +161,99 @@ def test_with_runs(self):
                         assert course_data['runs'][0]['status'] == CourseStatus.CURRENT_GRADE
                     if course_data['runs'][0]['course_id'] == "course-v1:MITx+8.MechCX+2014_T1":
                         assert course_data['runs'][0]['status'] == CourseStatus.UPGRADE
+
+
+class DashboardTokensTest(APITestCase):
+    """
+    Tests for access tokens in dashboard Rest API
+    """
+
+    @classmethod
+    def setUpTestData(cls):
+        super(DashboardTokensTest, cls).setUpTestData()
+        # create an user
+        cls.user = UserFactory.create()
+        # create a social auth for the user
+        cls.user.social_auth.create(
+            provider='edxorg',
+            uid=cls.user.username,
+            extra_data='{"access_token": "fooooootoken", "refresh_token": "baaaarrefresh"}'
+        )
+
+        cls.enrollments = Enrollments([])
+
+        # url for the dashboard
+        cls.url = reverse('dashboard_api')
+
+    def setUp(self):
+        super(DashboardTokensTest, self).setUp()
+        self.client.force_login(self.user)
+        self.now = datetime.now(pytz.utc)
+
+    def update_social_extra_data(self, data):
+        """Helper function to update the python social auth extra data"""
+        social_user = self.user.social_auth.get(provider='edxorg')
+        social_user.extra_data.update(data)
+        social_user.save()
+
+    @patch('backends.edxorg.EdxOrgOAuth2.refresh_token', autospec=True)
+    def test_refresh_token(self, mock_refresh):
+        """Test to verify that the access token is refreshed if it has expired"""
+        extra_data = {
+            "updated_at": (self.now - timedelta(weeks=1)).timestamp(),
+            "expires_in": 100  # seconds
+        }
+        self.update_social_extra_data(extra_data)
+
+        with patch(
+            'edx_api.enrollments.CourseEnrollments.get_student_enrollments',
+            autospec=True,
+            return_value=self.enrollments
+        ):
+            res = self.client.get(self.url)
```

Comment:
looks like everything from line 206 to this line can be moved into a helper method that returns `res`. the code in all 3 test cases is identical